k8s/object.bzl

```diff
@@ -96,9 +96,67 @@ EOF""".format(
     ctx.outputs.yaml,
   ] + all_inputs))
 
+def _common_impl(ctx, template):
+  kwargs = {
+      "cluster": ctx.attr.cluster,
+      "namespace": ctx.attr.namespace,
+      "script": ctx.outputs.executable.path,
+      "kind": ctx.attr.kind,
+  }
+  files = [ctx.executable._resolver]
+
+  if hasattr(ctx.executable, "resolved"):
+    kwargs["resolve_script"] = ctx.executable.resolved.short_path
+    files += [ctx.executable.resolved]
+    files += list(ctx.attr.resolved.default_runfiles.files)
+
+  if hasattr(ctx.files, "unresolved"):
+    kwargs["unresolved"] = ctx.files.unresolved[0].short_path
+    files += ctx.files.unresolved
+
+  ctx.action(
+      command = template.format(**kwargs),
```

Comment:
Would it make more sense to use something like: https://docs.bazel.build/versions/master/skylark/lib/actions.html#expand_template

To generate the templated file?

Reply:
I moved the string passed from each caller into a file, and now everything goes through `ctx.actions.expand_template` and shares a common implementation :)